### DISCUSSION TITLE
Config Help

### DISCUSSION THREAD

**Discussion by SZ87** [devcompl/TTController#154] [Configuration]
@MoshiMoshi0,

I'm having a doozy of a time figuring out how to properly configure my setup. To help, [here is my "Report"](https://pastebin.com/GKgNjKyP) as generated by TTController. I'm no stranger to JSON, but I'm simply drawing blanks.

Not sure how helpful this would be, but here's what I'm working with:

- 2x Riing Plus 12 (rear and top, exhaust)
- 1x Floe Riing 280 AIO w/ 2x Riing Plus 14 (front panel, intake)

That's 5 devices in total.

A couple things to note:

1. Under "Controllers" section, "Name" reads as "Riing Quad", but I only have Riing Plus fans. Or is that just the name the controller identifies itself as?
2. Assuming "PortId" refers the the physical port # on the controller itself, it appears to be:
   - 1 = AIO pump
   - 2 = Front rad fan top
   - 3 = Front rad fan bottom
   - 4 = Rear fan
   - 5 = Top fan

Also, the case I'm using (TT V150) has my GPU blowing directly against the bottom PSU shroud...mount...panel thingy, which seems to have become an unintended heat trap. Is there a way to account for this using your program? In other words, could I prioritize the speeds of all my fans (or maybe just the exhaust fans) based on GPU temps? Or better yet, could I set my exhaust fans to be dependent on GPU temps? Do you get what I'm saying? My sincerest apologies. It's late and I'm tired 😩.

Thank you very much in advance!

**Comment by MoshiMoshi0** [comment]
> Under "Controllers" section, "Name" reads as "Riing Quad", but I only have Riing Plus fans

Thermaltake started only using the latest controller even for older fans.

This should work, you have to edit `"CurvePoints"` for Front and Rear profiles to tweak the pwm curve.
Front is based on CPU sensor, rear on GPU sensor.

```json
{
  "Profiles": [
    {
      "Name": "Front",
      "Ports": [
        [9802, 8800, 2],
        [9802, 8800, 3]
      ],
      "SpeedControllers": [
        {
          "Type": "PwmSpeedController",
          "Config": {
            "CurvePoints": [
              [50, 20],
              [70, 30],
              [80, 50],
              [85, 75],
              [90, 100]
            ],
            "Sensors": ["/intelcpu/0/temperature/6"],
            "MinimumChange": 4,
            "MaximumChange": 8,
            "Trigger": {
              "Type": "AlwaysTrigger"
            }
          }
        }
      ]
    },
    {
      "Name": "Rear",
      "Ports": [
        [9802, 8800, 4],
        [9802, 8800, 5]
      ],
      "SpeedControllers": [
        {
          "Type": "PwmSpeedController",
          "Config": {
            "CurvePoints": [
              [50, 20],
              [70, 30],
              [80, 50],
              [85, 75],
              [90, 100]
            ],
            "Sensors": ["/gpu-nvidia/0/temperature/0"],
            "MinimumChange": 4,
            "MaximumChange": 8,
            "Trigger": {
              "Type": "AlwaysTrigger"
            }
          }
        }
      ]
    },
    {
      "Name": "RGB",
      "Ports": [
        [9802, 8800, 1],
        [9802, 8800, 2],
        [9802, 8800, 3],
        [9802, 8800, 4],
        [9802, 8800, 5]
      ],
      "Effects": [
        {
          "Type": "AuroraEffect",
          "Config": {
            "Step": 0.003,
            "Length": 64,
            "Mirror": false,
            "Trigger": {
              "Type": "AlwaysTrigger"
            }
          }
        }
      ]
    }
  ],
  "PortConfigs": [
    {
      "Ports": [
        [9802, 8800, 1]
      ],
      "Config": {
        "DeviceType": "FloeRiing",
        "IgnoreColorCache": "True"
      }
    }, 
    {
      "Ports": [
        [9802, 8800, 2],
        [9802, 8800, 3],
        [9802, 8800, 4],
        [9802, 8800, 5]
      ],
      "Config": {
        "IgnoreColorCache": "True"
      }
    }
  ]
}
```
